src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj

```diff
@@ -38,4 +38,73 @@
     <InternalsVisibleToTest Include="$(AssemblyName).Tests" />
     <InternalsVisibleToDynamicProxyGenAssembly2 Include="*" />
   </ItemGroup>
+
+  <ItemGroup>
```

Comment:
Our implementation of rewriting targets files for net462 makes us duplicate the content, and in this case it's a significant maintenance overhead.

I think we can be a little smarter and preserve the existing targets, though just rename it during the packaging.
This should work:
```diff
diff --git a/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj b/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
index f7b7fa6d85..0861d4261c 100644
--- a/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
+++ b/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
@@ -40,10 +40,17 @@
   </ItemGroup>
 
   <ItemGroup>
-    <None Include="buildTransitive\$(MSBuildProjectName).*"
+    <None Include="buildTransitive\**\*"
           CopyToOutputDirectory="PreserveNewest"
           Pack="true"
           PackagePath="buildTransitive\$(MinimumSupportedTfmForPackaging)\" />
+    <None Include="buildTransitive\Microsoft.Extensions.Http.Resilience.targets"
+          Condition="'$(MinimumSupportedTfmForPackaging)' != '' and
+                     '$(ConditionalNet462)' != '' and
+                     ($(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)')) and $(TargetFrameworks.Contains('$(ConditionalNet462)')))"
+          CopyToOutputDirectory="PreserveNewest"
+          Pack="true"
+          PackagePath="buildTransitive\net462\Microsoft.Extensions.Http.Resilience.net462.targets" />
     
     <!-- For net462 we copy only .props file. The .targets file is added automatically by the build process, and the code
          that checks the version of the Grpc.Net.ClientFactory package will be added to the .targets file as a value
@@ -59,51 +66,7 @@
   <PropertyGroup>
     <_AdditionalNETStandardCompatErrorFileContents>
       <![CDATA[
-  <!--
-    Check whether the project is referencing Grpc.Net.ClientFactory 2.64.0 or later.
-    If the vesion is earlier warn the user to update.
-    See https://github.com/dotnet/extensions/tree/main/src/Libraries/Microsoft.Extensions.Http.Resilience#known-issues for more details.
-    -->
-  <Target Name="_CheckGrpcNetClientFactoryVersion"
-          BeforeTargets="ResolveReferences"
-          Condition=" '%24(SuppressCheckGrpcNetClientFactoryVersion)' != 'true' ">
-    <ItemGroup>
-      <!-- Find the package in the .csproj file. -->
-      <_GrpcNetClientFactoryPackageReference Include="%40(PackageReference)" Condition=" '%(PackageReference.Identity)' == '%24(_GrpcNetClientFactory)' " />
-
-      <!-- Find the version of the package in the Central Package Source. The solution uses the Central Package Management. -->
-      <_GrpcNetClientFactoryPackageVersion Include="%40(PackageVersion)" Condition=" '%(PackageVersion.Identity)' == '%24(_GrpcNetClientFactory)' " />
-      
-      <!-- The package is added to the project as a transitive dependency. -->
-      <_GrpcNetClientFactoryTransitiveDependency Include="%40(ReferencePath)" Condition=" '%(ReferencePath.NuGetPackageId)' == '%24(_GrpcNetClientFactory)' " />
-    </ItemGroup>
-
-    <!-- The version of the package is included in the .csproj file. -->
-    <Error Condition=" %40(_GrpcNetClientFactoryPackageReference->Count()) &gt%3B 0
-                       AND '%(_GrpcNetClientFactoryPackageReference.Version)' != ''
-                       AND %24([MSBuild]::VersionLessThan('%(_GrpcNetClientFactoryPackageReference.Version)', '%24(_CompatibleGrpcNetClientFactoryVersion)')) "
-           Text="%24(_GrpcNetClientFactoryVersionIsIncorrect)" />
-
-    <!-- The solution uses the Central Package Management and the version of the package is overridden in the .csproj file using the VersionOverride property. -->
-    <Error Condition=" '%24(ManagePackageVersionsCentrally)' == 'true'
-                       AND %40(_GrpcNetClientFactoryPackageReference->Count()) &gt%3B 0
-                       AND '%(_GrpcNetClientFactoryPackageReference.VersionOverride)' != ''
-                       AND %24([MSBuild]::VersionLessThan('%(_GrpcNetClientFactoryPackageReference.VersionOverride)', '%24(_CompatibleGrpcNetClientFactoryVersion)')) "
-           Text="%24(_GrpcNetClientFactoryVersionIsIncorrect)" />
-
-    <!-- The solution uses the Central Package Management and the version of the package is included in the Central Package Source. -->
-    <Error Condition=" '%24(ManagePackageVersionsCentrally)' == 'true'
-                       AND %40(_GrpcNetClientFactoryPackageReference->Count()) &gt%3B 0
-                       AND '%(_GrpcNetClientFactoryPackageVersion.Version)' != ''
-                       AND %24([MSBuild]::VersionLessThan('%(_GrpcNetClientFactoryPackageVersion.Version)', '%24(_CompatibleGrpcNetClientFactoryVersion)')) "
-           Text="%24(_GrpcNetClientFactoryVersionIsIncorrect)" />
-
-    <!-- This condition handles a case when the package is added to the project as a transitive dependency. -->
-    <Error Condition=" %40(_GrpcNetClientFactoryTransitiveDependency->Count()) &gt%3B 0
-                       AND '%(_GrpcNetClientFactoryTransitiveDependency.NuGetPackageVersion)' != ''
-                       AND %24([MSBuild]::VersionLessThan('%(_GrpcNetClientFactoryTransitiveDependency.NuGetPackageVersion)', '%24(_CompatibleGrpcNetClientFactoryVersion)')) "
-           Text="%24(_GrpcNetClientFactoryVersionIsIncorrect)" />
-  </Target>
+  <Import Project="%24(MSBuildThisFileDirectory)\Microsoft.Extensions.Http.Resilience.net462.targets" />
 ]]>
     </_AdditionalNETStandardCompatErrorFileContents>
   </PropertyGroup>
```

![image](https://github.com/user-attachments/assets/ff1e6307-d9ff-4dd7-9579-68cdff750908)


Though, please test it.

Reply:
Thank you. Done. Tested and it worked.